src/pqueue.py

```diff
@@ -0,0 +1,47 @@
+class Pqueue(object):
+
+    def __init__(self, item):
+        self.pqueue = []
+        self.insert(item)
+
+    def insert(self, item):
+        for pair in item:
+            self.pqueue.append(pair)
+            self._sort_up()
+
+    def _sort_up(self):
+        idx = len(self.pqueue) - 1
+        while (idx-1)//2 >= 0:
+            if self.pqueue[idx][0] > self.pqueue[(idx-1)//2][0]:
+                temp_parent = self.pqueue[(idx-1)//2]
+                self.pqueue[(idx-1)//2] = self.pqueue[idx]
+                self.pqueue[idx] = temp_parent
+            idx = (idx-1)//2
```

Comment:
PEP8 E226 should be used for this class (whitespace around arithmetic operators). It has better readability:

```python
    def _sort_up(self):
        idx = len(self.pqueue) - 1
        while (idx - 1) // 2 >= 0:
            if self.pqueue[idx][0] > self.pqueue[(idx - 1) // 2][0]:
                temp_parent = self.pqueue[(idx - 1) // 2]
                self.pqueue[(idx - 1) // 2] = self.pqueue[idx]
                self.pqueue[idx] = temp_parent
            idx = (idx - 1) // 2
```

 Flake8 should be picking this up.